### ISSUE TITLE
Add background color behind text

### ISSUE THREAD

**Issue by sofroniewn** [napari/napari#3186]
## 🚀 Feature
Sometimes it is hard to read our text because the background can be uneven. Adding a background color to the textmanager and vispy text visual that filled in the bounding box behind to box could address this.

For example making something like

![image](https://user-images.githubusercontent.com/6531703/129659657-eeb49277-1141-483f-bb90-1c03bbd62b5c.png)

instead of just

![image](https://user-images.githubusercontent.com/6531703/129659906-6f2a2e21-109b-4c98-b138-ee6ac280518a.png)

This is likely something that would need to be contributed upstream to the vispy text visual. I was unable to find any discussions on the vispy github but @djhoese might have ideas or thoughts which are always appreciated.

Thoughts @kevinyamauchi @alisterburt too?

**Comment by alisterburt** [comment]
agree - the ideal would be stroked text I think, followed by a background as you suggest. I'll look into this a bit too

**Comment by kevinyamauchi** [comment]
I totally agree that giving the text a background color is necessary for the text to be legible on a varied background. In one of the early attempts at text, we tried overlaying two text visuals (background one with a larger font size), but the text size doesn't scale uniformly, so it didn't work. Perhaps there 

With regards to adding a background rectangle, I think we experimented with estimating the size of text elements in https://github.com/napari/napari/pull/600/. We could consider making a compound visual that combines a text visual with a mesh visual.

**Comment by djhoese** [comment]
I like the idea of stroked text, I see that there is some `distance` and `width` logic in the fragment shader in vispy:

https://github.com/vispy/vispy/blob/af2439895a13f187d840a476bd73ed480f2978f2/vispy/visuals/text/text.py#L377-L383

Maybe that can be used to "fake" the stroking to do a minimal working solution for it. I'm not familiar with all the stroking possibilities in CSS (the only thing that comes up when you google stroked text) so maybe this is a bad idea and not flexible enough but :shrug:

**Comment by sofroniewn** [comment]
Stroking sounds great! It would also be nice if we could avoid adding another visual for performance reasons. Thanks for discussion and ideas so far!!
